Docs: Added walkthrough documentation

Raxnam opened the pull request:
Added walkthrough documentation for each demo
TODO: still needs images and possible links to other docs

Commit:
Docs: Added walkthrough documentation

Added walkthrough documentation for each demo
TODO: still needs images

arthur740212 commented on docs/SceneWalkthrough/AchievementsWalkthrough.md:
```diff
@@ -0,0 +1,11 @@
+## **Achievements Demo**
+This demo showcases the Achievements function. This is done through 2 test achievements, one logging logins and the other logging an arbitrary stat. Each Achievement has a unlocked and locked display name and description, as well as the ability to hide it and the relevant tracked data for it.
```
Line 2 : logging logins confused a bit at first glance, maybe tracking login count could be better

arthur740212 commented on docs/SceneWalkthrough/Peer2PeerWalkthrough.md:
```diff
@@ -0,0 +1,8 @@
+## **Peer 2 Peer Demo**
+The Peer 2 Peer demo showcases the peer 2 peer interface. This is done thoru a basic messaging application where users can chat with other users on their friend list.
```
Line 2 : There is an incomplete word

andrew-hirata-playeveryware replied:
The spelling error "thoru" should read as through.

andrew-hirata-playeveryware commented on docs/SceneWalkthrough/AchievementsWalkthrough.md:
```diff
@@ -0,0 +1,11 @@
+- The ``Increase Login Count`` button increases the login counter used for the first achievement
+- The icon labeled 01 is the login count achievement, it can be selected to view its data
+- The icon labeled Max Stat 50 is the stat count achievement, it can be selected to view its data
+- The ``Unlock`` button unlocks the selected achievement, regardless of its current state
```
I feel for lists of full sentences, we should probably end them with periods.

andrew-hirata-playeveryware commented on docs/SceneWalkthrough/P2PNetcodeWalkthrough.md:
```diff
@@ -0,0 +1,11 @@
+
+## **P2P Netcode Demo**
+This demo showcases the peer 2 peer interface working with unity's netcode. This is done through a simple game where players can move a square around the screen
```
Capitalize 'Unity'

andrew-hirata-playeveryware commented on docs/SceneWalkthrough/P2PNetcodeWalkthrough.md:
```diff
@@ -0,0 +1,11 @@
+- The ``Join`` button in the friends tab allows the user to join a friend who is hosting a game
+
+
+❗ More documentation on the p2p interface can be found [here](https://dev.epicgames.com/docs/game-services/p-2-p)
```
Capitalize p2p as P2P.

andrew-hirata-playeveryware commented on docs/SceneWalkthrough/PerformanceStressTestWalkthrough.md:
````diff
@@ -0,0 +1,12 @@
+    - The ``Start GPU Test`` and ``Stop GPU Test`` start and stop the gpu usage tests respectively. This test does not have a target utilization, but rather renders a complex scene which should use most if not all of a graphics card's computational bandwidth
+- Memory Test
+    - The ``Start Memory Test`` and ``Stop Memory Test`` start and stop the memory usage tests respectively
+    - The ```Memory Allocation``` Slider determines how much memory to allocate for the test (Due to how unity handles memory, this memory is not returned to the system until the program is closed, however ending the test returns it to unity's memory manager)
````
Capitalize "unity"

andrew-hirata-playeveryware commented on docs/SceneWalkthrough/PlayerDataStorageWalkthrough.md:
```diff
@@ -0,0 +1,12 @@
+## **Player Data Storage Demo**
+This demo showcases the player data storage interface. This allows games to store player specific data in cloud servers. The demo uses a faux inventory to showcase this.
```
Capitalize "player data storage" to "Player Data Storage"

andrew-hirata-playeveryware commented on docs/SceneWalkthrough/PlayerReportsAndSanctionsWalkthrough.md:
```diff
@@ -0,0 +1,13 @@
+## **Player Reports and Sanctions Demo**
+This demo showcases the reports interface, and sanctions interface. This is done through a report menu, and a display of any current sanctions on the user's account
```
I think the comma after "interface" is unneeded. @RedMazer probably knows the answer.

andrew-hirata-playeveryware commented on docs/Walkthrough.md:
```diff
@@ -0,0 +1,76 @@
+## **EOS Plugin for Unity Walkthrough**
+ 
+This document is meant to serve as a guide to our EOS Plugin for Unity demo scenes. Each scene is meant to showcase the implementation of a specific feature such as peer 2 peer communication, lobbies, or custom invites.
+Each scene has a few things in common, the login page, side bar, scene selection, and debug log.
```
I thin the first comma should be a colon "... few things in common: the login page ..."

arthur740212 commented on docs/Walkthrough.md:
```diff
@@ -0,0 +1,76 @@
+
+
+## Scene selection
+The yellow dropdown in the dop of the window allows the user to select which demo scene they would like to log into
```
Line 28 : small typo

arthur740212 commented on docs/Walkthrough.md:
```diff
@@ -0,0 +1,76 @@
+
+
+## **Individual Scene Walkthroughs**
+- [Achievements](SceneWalkthrough\AchievementsWalkthrough)
```
Line 64 65 .md missing for the links

arthur740212 reviewed (commented):
Small typos and missing words.
The walkthrough files themselves seems in depth and accurate.

scovertb commented on docs/SceneWalkthrough/LobbiesWalkthrough.md:
```diff
@@ -0,0 +1,33 @@
+    - ``Allow Invites`` determines whether or not players in the lobby can invite other players
+    - ``Presence Enabled`` determines whether or not to associate the lobby with the [presence](https://dev.epicgames.com/docs/epic-account-services/eos-presence-interface) system
+    -  ``RTC Voice Room`` determines whether or not the lobby will have voice chat
+    - ``Anti-Cheat Enabled`` toggles Easy-Anti Cheat on and off
```
More precisely, enabling the EAC toggle will kick cheating players or leave if the owner is cheating

Commit:
Docs: Updated walkthrough documents

Fixed typos and some formatting issues, and updated info

RedMazer commented on docs/Walkthrough.md:
```diff
@@ -0,0 +1,71 @@
+The sidebar shows the current version, logged in account, framerate, readme, log copy button and exit button.
+ - The ``README`` button will display the readme for the current scene
+ - The ``Copy Log`` button copies the log from the log window
+ - The ``Exit11    button closes the application
```
looks like its missing a ```

RedMazer commented on docs/Walkthrough.md:
```diff
@@ -0,0 +1,71 @@
+- ``Connect``: Uses the connect interface to login through one of the supported options. More documentation on this can be found [here](https://dev.epicgames.com/docs/game-services/eos-connect-interface?sessionInvalidated=true).
+    - Device Access Token: Uses a persistent access token based on the device, unattached to an account.
+    - Steam Session Ticket
+    - Steam App Ticket
```
Steam Session Ticket
Steam App Ticket

just wanna confirm those are the correct terms. i have no idea.

Raxnam replied:
I just pulled those names from the demo so they're my best guess

RedMazer commented on docs/SceneWalkthrough/AchievementsWalkthrough.md:
```diff
@@ -0,0 +1,11 @@
+- The ``Unlock`` button unlocks the selected achievement, regardless of its current state.
+
+
+❗ More information on the Achievement interface can be found [here](https://dev.epicgames.com/docs/game-services/achievements).
```
would recomend replacing every instance of these ! in each doc with
`> :heavy_exclamation_mark:`
emoji name, and  quote thing will make it pop more

RedMazer commented on docs/SceneWalkthrough/AchievementsWalkthrough.md:
```diff
@@ -0,0 +1,11 @@
+- The icon labeled 01 is the login count achievement, it can be selected to view its data.
+- The icon labeled Max Stat 50 is the stat count achievement, it can be selected to view its data.
+- The ``Unlock`` button unlocks the selected achievement, regardless of its current state.
+
```
are we able to remove achievements from an account? a link to where to do that would be useful

Raxnam replied:
I think you can through the dev portal, but i dont know too much about how the dev portal works

Raxnam replied:
and the url for the achievements section of the dev portal is unique to the product so I cant link it

arthur740212 replied:
yes, I might be able to write a doc for that, just will have to be very careful not to show anything confidential from our portal

RedMazer replied:
that should all be in epic docs right? a link to those should suffice.
or honestly something like, you can mess with things on the dev portal [link to dev portal homepage] would be fine for our current purposes, i think

RedMazer commented on docs/SceneWalkthrough/Auth&FriendsWalkthrough.md:
```diff
@@ -0,0 +1,7 @@
+This demo showcases an implementation of the friends list into a game UI. The panel on the right shows the user's friend list and status of each friend.
+- The ``Refresh`` button refreshes the list, repolling for information.
+- The ``Friends Overlay`` button opens the EOS overlay.
+
```
a note that you can open and close the friends tab would be nice and applicable to other scenes

RedMazer replied:
i think this needs like a 
- The ``Friends`` tab, on the top left of the friends list, opens and closes the friends list. **In other samples it may start closed**

RedMazer commented on docs/SceneWalkthrough/CustomInvitesWalkthrough.md:
```diff
@@ -0,0 +1,10 @@
+This demo showcases the custom invite functionality allowing for messaging with arbitrary payloads. This is meant for cases where an already existing invitation system exists. This demo allows users to send invites with a text payload to anyone on their friends list.
+- The ``Payload`` field is where the text payload should be input.
+- The ``Set`` button finalizes the invitation creation allowing for it to be sent.
+- The ``Invite`` button next to any name on the friends list sends the currently set invitation to that user.
```
a more definitive note that a payload must be set for the invite button to look/be enabled would be nice
maybe a comment on then needing to wait or mash the refresh button if its still isnt enabled

RedMazer commented on docs/SceneWalkthrough/LobbiesWalkthrough.md:
```diff
@@ -0,0 +1,33 @@
+    - ``LobbyId`` Searches by the lobby id as found at the top of the lobby.
+    - ``Level`` Searches by level, as chosen by the host when creating the lobby.
+    - ``BucketId`` Searches by Bucket ID, as chosen by the host when creating the lobby.
+
```
should probably have one for our invite popup that happens in the top right (behind the friends list), and a not that you can open the friends list in ours to invite people, and or use it to open the social overlay to invite people

RedMazer commented on docs/SceneWalkthrough/PlayerDataStorageWalkthrough.md:
```diff
@@ -0,0 +1,12 @@
+- The ``Download`` button downloads the currently selected file to the disk.
+- The ``Add Item`` add the item from the dropdown to the current local inventory.
+- The ``Upload`` button uploads the current local inventory file to the cloud, overwriting the old version.
+
```
do you need to do things in the developer portal to make this work? if so might need a note on that

Raxnam replied:
from my understanding the client generates the file, the only options in the dev portal are to look at individual player's "uploads"

andrew-hirata-playeveryware replied:
I believe that's how it works.

RedMazer commented on docs/SceneWalkthrough/StoreWalkthrough.md:
```diff
@@ -0,0 +1,8 @@
+- There are two options, ``CaliforniumGeneralAudience``, and ``DevCaliforniumAudience``.
+- The ``Checkout`` buttons to the right open up the epic store check out page with the respective product in the cart
+- ``Query Offers`` refreshes the available items in the shop.
+
```
ive never actually seen this work, but im assuming something needs to be set up on the dev portal, could use a note and link on that

Raxnam replied:
I was looking for documentation on it, but there's basically none on how to actually use the dev portal, I only could find the SDK documentation

RedMazer replied:
yeah i cant really find anything either, theres a few docs like all over the place, i wonder if maybe the bottom of every doc here should say something like
'for more information on the specific features, you can check out the Epic docs here'
https://dev.epicgames.com/docs
that way we can kinda just blatantly wash our hands of any weird feature things that arent connected to the samples?

Commit:
Docs: Added fixes and more info

Fixed typos, formatting and added a bit more to a few scenes' walkthroughs

Commit:
Docs: Updates

Added achievement managment info
added more details to friends tab
added a general EOS docs link on the main page

Commit:
Docs: Added Title Storage docs

Commit:
Docs: Added login type by platform documentation

Added information and link to the platform login table

Commit:
Docs: Added links to login table

added links to the specific readme files for each platform and renamed files to match naming convention

andrew-hirata-playeveryware commented on docs/SceneWalkthrough/TitleStorageWalkthrough.md:
```diff
@@ -0,0 +1,16 @@
+## **Title Storage Demo**
+This demo showcases the title storage interface. This functions similar to the player data storage interface, although the data accessed is game/platform specific instead of player specific. This demo allows users to search files by tag and name and then view their contents. The demo is broken up into three parts, Tags search, File Search and the File viewer
```
Missing a period or colon on the final sentence.

RedMazer commented on docs/Walkthrough.md:
```diff
@@ -0,0 +1,75 @@
+    - Steam App Ticket
+    - Discord Access Token  
+    - Openid Access Token
+
```
This is gonna need an update stating which login type needs to be used on which platform, as well as which can be used  to access features that dont require an Epic account. altering that one table we have might be a good option here?

RedMazer replied:
i think docs/readme_login.md is the doc you want to link to in the !, if im understanding that right

Commit:
Docs: Added image for login type by platform

moved to using an image instead of a link to the table

andrew-hirata-playeveryware commented on Packages/manifest.json:
```diff
@@ -1,5 +1,6 @@
 {
   "dependencies": {
+    "com.rlabrecque.steamworks.net": "https://github.com/rlabrecque/Steamworks.NET.git?path=/com.rlabrecque.steamworks.net",
```
This shouldn't be a hard dependency in the Project.

arthur740212 commented on docs/SceneWalkthrough/AchievementsWalkthrough.md:
```diff
@@ -0,0 +1,28 @@
+
+
+During development, one might test whether a given achievement is functioning.
+To make testing easier, one could manually unloack or reset an user's achievement status through the developer portal
```
mispelled (unlock)

Commit:
Docs: Updated file names and added images

Renamed all files to match naming convention
Added images to relevant files
Removed steamworks package

Commit:
Docs: Additional images

Hadn't saved a few files
